src/main/java/org/spongepowered/api/util/weighted/WeightedCollection.java

```diff
@@ -56,22 +57,19 @@ public WeightedCollection() {
     /**
      * Selects a random value from this list based on their weight.
      * 
-     * <p>If the list is empty then null will be returned.</p>
-     * 
      * @param rand The random object to use for selection
-     * @return The selected value, or null if the list was empty
+     * @return The selected value, if the list is not empty
      */
-    public T get(Random rand) {
+    public Optional<T> get(Random rand) {
```

Comment:
Having it throw an exception, calling get on an empty weighted collection is exceptional.